addons/mail/static/src/core/messaging_service.js

```diff
@@ -635,6 +643,18 @@ export class Messaging {
         }
     }
 
+    _removeThread(threadData) {
+        const thread = this.threadService.insert({
+            ...threadData,
+            model: "discuss.channel",
+        });
+        this.threadService.remove(thread);
```

Comment:
`threadService.remove(thread)` should happen at the very last step of handling the notification. It's not okay to remove it from the store, and then have some code relying on `thread` object.